src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -665,6 +665,14 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
                 }
             }
 
+            // Account for the situation where we have trailing required parameters AFTER an optional one, i.e., (int opt1 = 3, T1 req1, ..., TN reqN) => {...}
+            // so we will be missing null default values for req1, ..., reqN when exiting the previous loop and the length of the default value builder will
+            // be too short.
+            if (parameterDefaultValueBuilder is { })
```

Comment:
Nit: prefer `not null` instead of `{}` if you don't need to create a variable. Don't let @jaredpar corrupt you.

Reply:
Actually I'd advocate for `is null` / `is not null`. I don't like `x is { }` because it has the flaw of being silently promoted to `true` when `x` is a `struct`. At this point I think `x is null` is equivalent to `x is object` for all the weird cases I cared about. The `x is {}` is the problem case: it's shorter but less precise :smile:

Reply:
Jared filled me in on the backstory of `x is { }` and I understand the pitfalls now so I will be changing this to `x is not null`